File: app/src/main/java/com/liadpaz/amp/adapters/ExtendedViewPagerAdapter.java
package com.liadpaz.amp.adapters;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import com.liadpaz.amp.livedatautils.QueueUtil;

public class ExtendedViewPagerAdapter extends FragmentStateAdapter {

    public ExtendedViewPagerAdapter(@NonNull Fragment fragment) {
        super(fragment);
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        return null; //ExtendedSongFragment.newInstance(position);
    }

    @Override
    public int getItemCount() {
        return QueueUtil.getQueueSize();
    }
}

File: app/src/main/java/com/liadpaz/amp/ui/adapters/ExtendedViewPagerAdapter.kt
package com.liadpaz.amp.ui.adapters

import androidx.fragment.app.Fragment
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.liadpaz.amp.ui.fragments.ExtendedSongFragment.Companion.newInstance
import com.liadpaz.amp.livedatautils.QueueUtil

class ExtendedViewPagerAdapter(fragment: Fragment) : FragmentStateAdapter(fragment) {
    private val TAG = "AmpApp.ExtendedViewPagerAdapter"

    override fun createFragment(position: Int): Fragment {
        return newInstance(position)
    }

    override fun getItemCount(): Int = QueueUtil.queueSize
}